GridKit/AutomaticDifferentiation/DependencyTracking/Variable.hpp

```diff
@@ -254,6 +254,12 @@ namespace GridKit
       inline Variable& operator/=(const double& rhs);
       inline Variable& operator/=(const Variable& rhs);
 
+      // conversion operator
+      inline operator double() const
+      {
+        return value_;
+      }
```

Comment:
Why not call the function `value()`?

```suggestion
      inline operator value() const
      {
        return value_;
      }
```
It will not necessarily return a double floating precision type.